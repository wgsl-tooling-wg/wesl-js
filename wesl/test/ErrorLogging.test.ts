import { type expect, test } from "vitest";
import { linkWithLogQuietly } from "./TestUtil.ts";
import { assertSnapshot } from "@std/testing/snapshot";

test("unresolved identifier", async (t) => {
  const src = `
    fn main() { x = 7; }
    `;
  const { log } = await linkWithLogQuietly(src);
  await assertSnapshot(t, log);
});

test("conditionally empty struct", async (t) => {
  const src = `
    struct Empty {
      @if(false) u: u32,
    }
    `;
  const { log } = await linkWithLogQuietly(src);
  await assertSnapshot(t, log);
});
