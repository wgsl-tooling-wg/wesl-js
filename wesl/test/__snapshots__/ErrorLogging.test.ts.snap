export const snapshot = {};

snapshot[`unresolved identifier 1`] = `
"unresolved identifier 'x' in file: ./test.wesl
    fn main() { x = 7; }   Ln 2
                ^^"
`;

snapshot[`conditionally empty struct 1`] = `
"struct Empty in ./test.wesl has no members (with current conditions)
    struct Empty {   Ln 2
    ^"
`;
